src/Mvc/Mvc.ViewFeatures/src/Infrastructure/ComponentHtmlContent.cs

```diff
@@ -1,26 +1,30 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Text.Encodings.Web;
 using Microsoft.AspNetCore.Html;
-using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures
 {
     internal class ComponentHtmlContent : IHtmlContent
     {
+        private readonly IEnumerable<string> _preamble;
         private readonly IEnumerable<string> _componentResult;
+        private readonly IEnumerable<string> _epilogue;
 
         public ComponentHtmlContent(IEnumerable<string> componentResult)
-        {
-            _componentResult = componentResult;
-        }
+            : this(Array.Empty<string>(), componentResult, Array.Empty<string>()) { }
+
+        public ComponentHtmlContent(IEnumerable<string> preamble, IEnumerable<string> componentResult, IEnumerable<string> epilogue) =>
+            (_preamble, _componentResult, _epilogue) = (preamble, componentResult, epilogue);
```

Comment:
why is this style better than braces and statements? Does it generate the same code?

I'm seeing this show up more and more as you touch things, and I'm not sure it's without tradeoffs.

Reply:
It's more succinct and it makes it obvious that you are just bulk assigning the parameters to fields. I think its more idiomatic when used with small objects like this one. I normally use it for readonly structs/types and DTOs.

ILSpy decompiles it to just assignments.

This is the IL it produces if you are curious. As you can see there's no actual tuple involved here:
```
.method public hidebysig specialname rtspecialname 
        instance void  .ctor(class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> preamble,
                             class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> componentResult,
                             class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> epilogue) cil managed
{
  // Code size       35 (0x23)
  .maxstack  2
  .locals init (class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> V_0,
           class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> V_1,
           class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> V_2)
  IL_0000:  ldarg.0
  IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
  IL_0006:  nop
  IL_0007:  ldarg.1
  IL_0008:  stloc.0
  IL_0009:  ldarg.2
  IL_000a:  stloc.1
  IL_000b:  ldarg.3
  IL_000c:  stloc.2
  IL_000d:  ldarg.0
  IL_000e:  ldloc.0
  IL_000f:  stfld      class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> Microsoft.AspNetCore.Mvc.ViewFeatures.ComponentHtmlContent::_preamble
  IL_0014:  ldarg.0
  IL_0015:  ldloc.1
  IL_0016:  stfld      class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> Microsoft.AspNetCore.Mvc.ViewFeatures.ComponentHtmlContent::_componentResult
  IL_001b:  ldarg.0
  IL_001c:  ldloc.2
  IL_001d:  stfld      class [System.Runtime]System.Collections.Generic.IEnumerable`1<string> Microsoft.AspNetCore.Mvc.ViewFeatures.ComponentHtmlContent::_epilogue
  IL_0022:  ret
} // end of method ComponentHtmlContent::.ctor
```